napari/utils/colormaps/colormap.py

```diff
@@ -294,47 +316,77 @@ def _cast_labels_to_minimum_dtype_auto(
     np.ndarray
         Casted labels data.
     """
+    data = _convert_small_ints_to_unsigned(data)
+
+    if data.itemsize <= 2:
+        return data
```

Comment:
This is not a big deal, but the flow is a bit weird here — `_convert_small_ints_to_unsigned` is a no-op for large dtypes, but then we are going to check for small dtypes anyway. Why not (a) make `_..._small_ints` work only with small ints, and then do:

```suggestion
    if data.itemsize <= 2:
        return _convert_small_ints_to_unsigned(data)
```

And I noticed that the function was used in another part of the code — I think it would be good to make all such mappings go through here, rather than shortcut to `_convert_small_ints_to_unsigned` for small dtypes.